typing/typecore.ml

```diff
@@ -4984,82 +5047,132 @@ and type_format loc str env =
   with Failure msg ->
     raise (Error (loc, env, Invalid_format msg))
 
-and type_label_exp create env loc ty_expected
```

Comment:
The changes here are extensive.
It will take me a while to review that.
Any information on what was done and why would certainly be helpful.